src/gpu/intel/jit/dsl/dsl.hpp

```diff
@@ -0,0 +1,270 @@
+/*******************************************************************************
+* Copyright 2025 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef GPU_INTEL_JIT_DSL_DSL_HPP
+#define GPU_INTEL_JIT_DSL_DSL_HPP
+
+#include <stack>
+
+#include "gpu/intel/jit/ir/ir.hpp"
+#include "gpu/intel/jit/ir/ir_builder.hpp"
+#include "gpu/intel/jit/pass/pass.hpp"
+#include "gpu/intel/jit/v2/conv/plan.hpp"
+#include "gpu/intel/jit/v2/ir/tensor.hpp"
+#include "ngen_core.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace gpu {
+namespace intel {
+namespace jit {
+namespace dsl {
+
+// TODO: Replace with hardware
+const int grf_size = 64;
+const int min_align_2d = 16;
+const int min_pitch_2d = 64;
```

Comment:
We should remove these before merging. These should be fairly easy to remove at this point as we can perform queries against `dsl::ctx_t::ctx_::hw()` within